### PR TITLE
Issue  #8650 fixed

### PR DESCRIPTION
**Pull request by malgamves**
Fixed the system error mentioned in issue #8650

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [x] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_agg.py [hunk 1]
```diff
@@ -161,8 +161,10 @@ def draw_path(self, gc, path, transform, rgbFace=None):
                 except OverflowError:
                     raise OverflowError("Exceeded cell block limit (set 'agg.path.chunksize' rcparam)")
         else:
-            try:
-                self._renderer.draw_path(gc, path, transform, rgbFace)
+            if rgbFace is not None:
+                rgbFace = tuple(rgbFace)
+            self._renderer.draw_path(gc, path, transform, rgbFace)
```
Comment (Kojoley) [inline lib/matplotlib/backends/backend_agg.py]: You accidentally removed `try`

Comment (malgamves) [reply lib/matplotlib/backends/backend_agg.py]: Let me fix that, thanks!

**Comment by tacaswell** [comment]
The circle CI failures are because the commit this is based on is older than the fix for that ( #9325 ), those can be disregarded here (or @malgamves can you rebase on to current master?)

The failure on travis is due to some trailing whitespace.  The automated style checkers are a bit annoying, but the do help keep the code-base consistent, which is important for big projects with lots of contributors.

It would be better if we had a test to exercise the original issue (to make sure this really fixes it!) but this code path is covered in the tests so I am :+1: to this going in with out an extra test.

Thanks for your work on this @malgamves !

**Comment by Kojoley** [comment]
I do not think the fix is doing a right thing. In many places you can find `rgbFace = mcolors.to_rgba(...)` or accessing the parts of rgbFace like that https://github.com/matplotlib/matplotlib/blob/9ea08fb22d975ea37075d6226827d5caeda1979c/lib/matplotlib/patches.py#L548

**Comment by malgamves** [comment]
Yes, I can rebase on the current master. Lets see if that works.

**Comment by anntzer** [comment]
I agree with @Kojoley that we should first figure out how a non-tuple made it to that point to start with (for example, it is relevant for mplcairo to know whether we can expect colorspecs to be normalized (hopefully yes) or not), and thus request a MWE from the OP of #8650.

**Comment by tacaswell** [comment]
This is the last step before we drop down to the c++ layer, I suspect that not-tuple may be the correct type for the python side and tuple is the correct type for this call.

**Comment by anntzer** [comment]
The culprit is
https://github.com/matplotlib/matplotlib/blob/57c845b158df87a23635b39447647b8e404cbdbf/src/py_converters.cpp#L198
which uses PyArg_ParseTuple (which expects exactly a tuple).

Note that this is also used by draw_markers (https://github.com/matplotlib/matplotlib/blob/master/src/_backend_agg_wrapper.cpp#L254) so again we should either make sure to normalize the input early, or fix convert_rgba to accept any sequence.

**Comment by jklymak** [comment]
I'm going to close this - it fixes a poorly documented error, and no one else has complained. There also seems to be some petered-out concern w/ how this was implemented.  Thanks @malgamves and sorry this didn't come to fruition.

**Pull request closed without merging**